appserver/payara-appserver-modules/microprofile/healthcheck/src/main/java/fish/payara/microprofile/healthcheck/HealthCheckService.java

```diff
@@ -92,18 +97,28 @@
     @Inject
     MetricsHealthCheckConfiguration configuration;
 
+    private boolean backwardCompEnabled;
+
     private static final Logger LOG = Logger.getLogger(HealthCheckService.class.getName());
 
-    private final Map<String, Set<HealthCheck>> healthChecks = new ConcurrentHashMap<>();
+    private final Map<String, Set<HealthCheck>> health = new ConcurrentHashMap<>();
+    private final Map<String, Set<HealthCheck>> readiness = new ConcurrentHashMap<>();
+    private final Map<String, Set<HealthCheck>> liveness = new ConcurrentHashMap<>();
+
     private final Map<String, ClassLoader> applicationClassLoaders = new ConcurrentHashMap<>();
     private final List<String> applicationsLoaded = new CopyOnWriteArrayList<>();
 
+    private static final String BACKWARD_COMP_ENABLED_PROPERTY = "MP_HEALTH_BACKWARD_COMPATIBILITY_ENABLED";
+
     @PostConstruct
     public void postConstruct() {
         if (events == null) {
             events = Globals.getDefaultBaseServiceLocator().getService(Events.class);
         }
         events.register(this);
+        this.backwardCompEnabled = ConfigProvider.getConfig()
```

Comment:
Should this also be checked on each invocation?
As it is, it's not dynamic which slightly defeats one of the purposes of MP Config.

Reply:
I purposely added this snippet to PostConstruct method instead of each invocation for very minor optimization. Do you think, end-user will need to change `MP_HEALTH_BACKWARD_COMPATIBILITY_ENABLED` property value dynamically?

Reply:
In 95% of cases, probably not.
But it's the fact that by being only checked in the postconstruct it essentially means that a restart is required if you want to change this. We can't however actually indicate that, because since it's MP config the option can be defined outside of Payara, so we have no way to check if it has changed except by, well, running this exact check :P

It's _probably_ fine as is, we'd just need to make sure to explicitly document this case.